test-case-reporting/app.ts

```diff
@@ -23,23 +23,26 @@ import express from 'express';
 import fs from 'fs';
 import statusCodes from 'http-status-codes';
 
+type Renderable = {builds: Array<Build>} | {testRuns: Array<TestRun>};
+
 const MAX_PAGE_SIZE = 500;
 const DEFAULT_PAGE_SIZE = 100;
+const TEMPLATE_DIR = './static';
 
 const app = express();
 const jsonParser = bodyParser.json({limit: '15mb'});
 const db = dbConnect();
 const record = new TestResultRecord(db);
 
-let testRunListTemplate = '';
-function renderTestRunList(testRuns: Array<TestRun>): string {
-  if (!testRunListTemplate) {
-    testRunListTemplate = fs
-      .readFileSync('./static/test-run-list.html')
+const templateCache: Record<string, string> = {};
+function render(templateName: string, data: Renderable): string {
```

Comment:
Tbh I'd just use `Object` here. More flexible when you start adding to the page context, and the typechecking is less essential since you'll quickly see the results in a rendered template if you screwed something up. We can always declare types in the view itself, before passing to the render function.

One alternative which I don't support but will list here for completeness would be defining wrappers for renderTestRunList and renderBuildList, which call render with the proper templates and have stricter types. I don't think that's cleaner or helpful though.

Reply:
Typescript complains when I use the type `Object`, telling me to use `Record<string, any>`, but then it complains about `any`.

How do you feel about making the type `Renderable = Record<string, (Renderable | string)>`?

Reply:
Use unknown instead of any

Reply:
`Record<string, unknown>`

Reply:
Done.